components/Helpers/src/ThemeListenerHelperWindow.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Win32;
+
+namespace CommunityToolkit.WinUI.HelpersRns;
+
+#if WINAPPSDK
+
+internal class ThemeListenerHelperWindow
+{
+    public delegate void ThemeChangedHandler(ApplicationTheme theme);
+
+    public static ThemeListenerHelperWindow Instance = new();
+    private static string s_className = "ThemeListenerHelperWindow";
+    private IntPtr m_hwnd;
+    public event ThemeChangedHandler? ThemeChanged;
+    delegate Windows.Win32.Foundation.LRESULT WndProc(Windows.Win32.Foundation.HWND hWnd, uint msg, Windows.Win32.Foundation.WPARAM wParam, Windows.Win32.Foundation.LPARAM lParam);
+    private WndProc m_wnd_proc_delegate;
+    private void registerClass()
+    {
+        unsafe
+        {
+            var wcx = new Windows.Win32.UI.WindowsAndMessaging.WNDCLASSEXW();
+            wcx.cbSize = (uint)Marshal.SizeOf(wcx);
+            fixed (char* pClassName = s_className)
+            {
+                wcx.lpszClassName = new Windows.Win32.Foundation.PCWSTR(pClassName);
+            }
+            wcx.lpfnWndProc = wndProc;
```

Comment:
This is not AOT safe. You need to use the blittable mode and pass a pointer to an `[UnsafeCallersOnly]` method.

Reply:
Use:

```c#
var pWindProc = Marshal.GetFunctionPointerForDelegate(wndProc);
var pfnWndProc = (delegate* unmanaged[Stdcall]<HWND, uint, WPARAM, LPARAM, LRESULT>)pWindProc;
```

Define delegate:
```c#
[UnmanagedFunctionPointer(CallingConvention.Winapi)]
public delegate LRESULT WNDPROC(HWND hWnd, uint msg, WPARAM wParam, LPARAM lParam);
```